src/overlays/actors/ovl_En_Dnk/z_en_dnk.c

```diff
@@ -374,127 +384,132 @@ void func_80A51CB8(EnDnk* this, PlayState* play) {
+                Math_SmoothStepToS(&this->currentHeadRot.y, headRot.y, 4, 0x1FFE, 1);
+                Math_SmoothStepToS(&this->currentHeadRot.z, headRot.z, 4, 0x1FFE, 1);
             } else {
-                this->unk_290 = sp3C.x;
-                this->unk_292 = sp3C.y;
-                this->unk_294 = sp3C.z;
+                this->currentHeadRot.x = headRot.x;
+                this->currentHeadRot.y = headRot.y;
+                this->currentHeadRot.z = headRot.z;
             }
         } else {
-            this->unk_28C |= 0x10;
-            this->unk_290 = sp3C.x;
-            this->unk_292 = sp3C.y;
-            this->unk_294 = sp3C.z;
+            this->flags |= 0x10;
+            this->currentHeadRot.x = headRot.x;
+            this->currentHeadRot.y = headRot.y;
+            this->currentHeadRot.z = headRot.z;
         }
 
-        Matrix_RotateYS(this->unk_292, MTXMODE_APPLY);
-        Matrix_RotateXS(this->unk_290, MTXMODE_APPLY);
-        Matrix_RotateZS(this->unk_294, MTXMODE_APPLY);
+        Matrix_RotateYS(this->currentHeadRot.y, MTXMODE_APPLY);
+        Matrix_RotateXS(this->currentHeadRot.x, MTXMODE_APPLY);
+        Matrix_RotateZS(this->currentHeadRot.z, MTXMODE_APPLY);
     }
 
     OPEN_DISPS(play->state.gfxCtx);
 
     MATRIX_FINALIZE_AND_LOAD(POLY_OPA_DISP++, play->state.gfxCtx);
-    gSPDisplayList(POLY_OPA_DISP++, this->unk_260[limbIndex]);
+    gSPDisplayList(POLY_OPA_DISP++, this->limbGfx[limbIndex]);
 
     CLOSE_DISPS(play->state.gfxCtx);
 }
 
-void func_80A51FC0(EnDnk* this, PlayState* play) {
+void EnDnk_Nuts_Draw(EnDnk* this, PlayState* play) {
     Gfx_SetupDL25_Opa(play->state.gfxCtx);
-    SkelAnime_DrawOpa(play, this->skelAnime.skeleton, this->skelAnime.jointTable, EnDnk_OverrideLimbDraw1,
-                      EnDnk_PostLimbDraw1, &this->actor);
+    SkelAnime_DrawOpa(play, this->skelAnime.skeleton, this->skelAnime.jointTable, EnDnk_Nuts_OverrideLimbDraw,
+                      EnDnk_Nuts_PostLimbDraw, &this->actor);
 }
 
-void func_80A52018(Actor* thisx, PlayState* play) {
+void EnDnk_Draw(Actor* thisx, PlayState* play) {
     EnDnk* this = (EnDnk*)thisx;
 
-    switch (ENDNK_GET_3(thisx)) {
-        case ENDNK_GET_3_0:
-            func_80A51CB8(this, play);
+    switch (ENDNK_GET_TYPE(thisx)) {
+        // palace guard type skeleton
+        case ENDNK_GET_TYPE_GUARD:
+            EnDnk_Guard_Draw(this, play);
             break;
 
-        case ENDNK_GET_3_1:
-        case ENDNK_GET_3_2:
-            func_80A51FC0(this, play);
+        // deku nuts type skeleton
+        case ENDNK_GET_TYPE_HINT:
+        case ENDNK_GET_TYPE_ATTACK:
+            EnDnk_Nuts_Draw(this, play);
             break;
 
         default:
             break;
     }
 }
 
-void func_80A52074(EnDnk* this, PlayState* play) {
+void EnDnk_HandleCSAudio(EnDnk* this, PlayState* play) {
     switch (play->csCtx.curFrame) {
-        case 80:
+        case 4 * 20:
```

Comment:
Why the * 20? Is there something special about the number 20?

Reply:
I'm attempting to group frames into seconds as it helps me to think about cutscenes in time rather than draw frames, trying to see if there is a pattern.
Although here, the frames are more useful in the end, as almost every odd-ball frame count is quarter second increment  minus 2 frames, where the pattern seems to be that the author wanted to pre-delay some sound effects to match the timing.